### DISCUSSION TITLE
Connecting an Unsupported Roaster That is Controlled by UART

### DISCUSSION THREAD

**Discussion by Couchy123** [artisan-roaster-scope/artisan#1769] [General]
What is the best way to connect an unsupported roaster?

I have a Kaleido Sniper M1, my version is not compatible with Artisan. It has a serial interface/screen that it communicates with through UART. 

What's the best way to combine Artisan and this UART communication?

**Comment by poundy** [comment]
Unless the protocol is documented and/or open, not really any way to incorporate inbuilt capability into another platform. Was there a reason you didn't go with the Artisan-ready model ?

**Reply by Couchy123** [reply]
It was quite a bit more for that one, and I thought I could work a way around, eventually.

I believe I have documented all the necessary information from the roaster. It's completely standard UART, and the outputs are even in ascii format. I have stored a list of all controller inputs and confirmed they all work.

How easy would it be to connect to Artisan with this information? Would I need a microcrontoller to work as  a middle man, decoding serial from both sides and translating them?

**Reply by poundy** [reply]
Your thinking about a box-in-the-middle was what I was thinking as well. You would need to emulate something on one side that Artisan can talk to, and then transform those to whatever serial comms you need. TC4 is the UART based controller that Artisan has a simple interface to, but there could be others that would work - you could try as a modbus controller as well.

**Reply by Couchy123** [reply]
okay great I'll give that a go. Thanks for the help!

**Reply by poundy** [reply]
TC4 is reasonably well known so shouldn't be too hard to bend to your preferred platform should you choose - I can't recall seing anyone share modbus controller code for an Arduino or similar that's been used here...
